aggregator/src/main/scala/com/stratio/sparkta/aggregator/DataCube.scala

```diff
@@ -65,50 +63,19 @@ case class DataCube(dimensions: Seq[Dimension],
         dimension <- dimensions
         value <- event.keyMap.get(dimension.name).toSeq
         (bucketType, bucketedValue) <- dimension.bucketer.bucket(value)
-      } yield DimensionValue(DimensionBucket(dimension, bucketType), convertToBucketTypeOp(bucketType,bucketedValue))
+      } yield DimensionValue(DimensionBucket(dimension, bucketType),
+          TypeOp.transformValueByTypeOp(bucketType.typeOp, bucketedValue))
       val eventTime = extractEventTime(dimensionValues)
       (DimensionValuesTime(dimensionValues, eventTime), event.keyMap)
     }).cache()
   }
 
-  //scalastyle:off
-  private def convertToBucketTypeOp(bucketType : BucketType, bucketedValue : JSerializable): JSerializable = {
-    bucketType.typeOp match {
-      case TypeOp.String => bucketedValue match {
-        case value if value.isInstanceOf[String] => value
-        case value if value.isInstanceOf[Seq[Any]] =>
-          value.asInstanceOf[Seq[Any]].mkString(Output.Separator).asInstanceOf[JSerializable]
-        case _ => bucketedValue.toString.asInstanceOf[JSerializable]
-      }
-      case TypeOp.ArrayDouble => bucketedValue match {
-        case value if value.isInstanceOf[Seq[Double]] => value
-        case value if value.isInstanceOf[Seq[Any]] =>
-          Try(value.asInstanceOf[Seq[Any]].map(_.toString.toDouble)).getOrElse(Seq()).asInstanceOf[JSerializable]
-        case _ => Try(Seq(bucketedValue.toString.toDouble)).getOrElse(Seq()).asInstanceOf[JSerializable]
-      }
-      case TypeOp.ArrayString => bucketedValue match {
-        case value if value.isInstanceOf[Seq[String]] => value
-        case value if value.isInstanceOf[Seq[Any]] =>
-          Try(value.asInstanceOf[Seq[Any]].map(_.toString)).getOrElse(Seq()).asInstanceOf[JSerializable]
-        case _ => Try(Seq(bucketedValue.toString)).getOrElse(Seq()).asInstanceOf[JSerializable]
-      }
-      case TypeOp.Timestamp => bucketedValue match {
-        case value if value.isInstanceOf[Long] => DateOperations.millisToTimeStamp(value.asInstanceOf[Long])
-        case value if value.isInstanceOf[Date] => DateOperations.millisToTimeStamp(value.asInstanceOf[Date].getTime)
-        case value if value.isInstanceOf[DateTime] => DateOperations.millisToTimeStamp(value.asInstanceOf[DateTime].getMillis)
-        case _ => Try(Seq(bucketedValue.toString)).getOrElse(Seq()).asInstanceOf[JSerializable]
-      }
-      case _ => bucketedValue
-    }
-  }
-  //scalastyle:on
-
   private def extractEventTime(dimensionValues: Seq[DimensionValue]) = {
```

Comment:
Extract this to a helper and test it. t is not necessary now but I think that it will be useful: Up to you!